syntax = "proto3";

package cart;

service CartService {
  rpc GetCart (GetCartRequest) returns (CartResponse);
  rpc CreateCart (CreateCartRequest) returns (CreateCartResponse);
}

message Cart {
  string id = 1;
  string product_id = 2;
  int32 quantity = 3;
}

message GetCartRequest {
  string id = 1;
  repeated string fields = 2;
}

message CreateCartRequest {
  string id = 1;
  string product_id = 2;
  int32 quantity = 3;
}

message CartResponse {
  Cart cart = 1;
}

message CreateCartResponse {
  string message = 1;
}
